### PR TITLE
entry ORDER BYにindex使えるgenerated columnを利用

### PR DESCRIPTION
**Pull request by cureseven**
#1
```
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]:         /home/isucon/isucon6-20230708/go/isuda.go:106 +0x805
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]: main.prepareHandler.func1({0x94a5c0, 0xc00021e0e0}, 0xc0003e5000)
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]:         /home/isucon/isucon6-20230708/go/util.go:18 +0xed
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]: main.myHandler.func1({0x94a5c0, 0xc00021e0e0}, 0xc0003004b0?)
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]:         /home/isucon/isucon6-20230708/go/util.go:31 +0x92
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]: net/http.HandlerFunc.ServeHTTP(0xc0003e4f00?, {0x94a5c0?, 0xc00021e0e0?}, 0xc0003279e8?)
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]:         /snap/go/current/src/net/http/server.go:2122 +0x2f
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]: github.com/gorilla/mux.(*Router).ServeHTTP(0xc000172000, {0x94a5c0, 0xc00021e0e0}, 0xc0003e4e
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]:         /home/isucon/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]: net/http.serverHandler.ServeHTTP({0xc000300360?}, {0x94a5c0, 0xc00021e0e0}, 0xc0003e4e00)
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]:         /snap/go/current/src/net/http/server.go:2936 +0x316
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]: net/http.(*conn).serve(0xc0003e02d0, {0x94a848, 0xc0001714a0})
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]:         /snap/go/current/src/net/http/server.go:1995 +0x612
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]: created by net/http.(*Server).Serve
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]:         /snap/go/current/src/net/http/server.go:3089 +0x5ed
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]: sql: Scan error on column index 1, name "keyword": converting driver.Value type []uint8 ("平
Jul 08 13:50:19 ip-172-31-34-154 isuda[8247]: runtime/debug.Stack()
```

### CODE REVIEW

File: go/isuda.go [hunk 1]
```diff
@@ -94,7 +93,7 @@ func topHandler(w http.ResponseWriter, r *http.Request) {
 	page, _ := strconv.Atoi(p)
 
 	rows, err := db.Query(fmt.Sprintf(
-		"SELECT * FROM entry ORDER BY updated_at DESC LIMIT %d OFFSET %d",
+		"SELECT id, author_id, ``keyword``, description, updated_at, created_at FROM entry ORDER BY updated_at DESC LIMIT %d OFFSET %d",
```
Comment (mism-mism) [inline go/isuda.go]: ここが間違ってる気がする

File: go/isuda.go [hunk 2]
```diff
@@ -92,7 +91,7 @@ func topHandler(w http.ResponseWriter, r *http.Request) {
...
```
Comment (cureseven) [inline go/isuda.go]:
```suggestion
		"SELECT id, author_id, \`keyword\`, description, updated_at, created_at FROM entry ORDER BY updated_at DESC LIMIT %d OFFSET %d",
```

**Comment by cureseven** [comment]
```
ALTER TABLE entry ADD COLUMN keyword_length INT UNSIGNED AS (CHARACTER_LENGTH(keyword)) STORED NOT NULL AFTER description;
ALTER TABLE entry ADD INDEX(keyword_length);
```

before
```
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+----------------+
| id | select_type | table | partitions | type | possible_keys | key  | key_len | ref  | rows | filtered | Extra          |
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+----------------+
|  1 | SIMPLE      | entry | NULL       | ALL  | NULL          | NULL | NULL    | NULL | 5596 |   100.00 | Using filesort |
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+----------------+
```
after
```
mysql> explain SELECT * FROM entry ORDER BY keyword_length DESC limit 500;
+----+-------------+-------+------------+-------+---------------+----------------+---------+------+------+----------+-------+
| id | select_type | table | partitions | type  | possible_keys | key            | key_len | ref  | rows | filtered | Extra |
+----+-------------+-------+------------+-------+---------------+----------------+---------+------+------+----------+-------+
|  1 | SIMPLE      | entry | NULL       | index | NULL          | keyword_length | 4       | NULL |  500 |   100.00 | NULL  |
+----+-------------+-------+------------+-------+---------------+----------------+---------+------+------+----------+-------+
```

**Comment by cureseven** [comment]
`{"pass":true,"score":6623,"success":2200,"fail":0,"messages":[]}`

**Pull request merged**